src/Console/Commands/JsonApiGeneratorCommand.php

```diff
@@ -0,0 +1,176 @@
+        $this->useEloquent = config('json-api.generator.use_eloquent', true);
+    }
+
+    /**
+     * Build the class with the given name.
+     *
+     * Remove the base controller import if we are already in base namespace.
+     *
+     * @param  string  $name
+     * @return string
+     */
+    protected function buildClass($name)
+    {
+        $namespace = $this->getNamespace($name);
+
+        $stub = $this->files->get($this->getStub());
+
+        $this->replaceNamespace($stub, $name)
+                ->replaceResourceType($stub, $this->getPluralizedResource());
+
+        return $stub;
+    }
+
+    /**
+     * Replace the value of the resource type constant
+     *
+     * @param mixed $stub
+     * @param mixed $resource
+     */
+    protected function replaceResourceType(&$stub, $resource)
+    {
+        $stub = str_replace(
+            'dummyResourceType', camel_case($resource), $stub
+        );
+
+        return $this;
+    }
+
+    /**
+     * Get the plural version of the resource name
+     *
+     * @return string
+     */
+    protected function getPluralizedResource()
+    {
+        return str_plural($this->getResourceName());
+    }
+
+    /**
+     * Get the resource name
+     *
+     * @return string
+     */
+    protected function getResourceName()
+    {
+        return ucwords($this->argument('resource'));
+    }
+
+    /**
+     * Get the desired class name from the input.
+     *
+     * @return string
+     */
+    protected function getNameInput()
+    {
+        return trim($this->type);
+    }
+
+    /**
+     * Get the stub file for the generator.
+     *
+     * @return string
+     */
+    protected function getStub()
+    {
+        if($this->isIndependant) {
+            return __DIR__.'/stubs/'.strtolower($this->type).'.stub';
+        }
+
+        if ($this->option('eloquent') || $this->useEloquent) {
+            return __DIR__.'/stubs/eloquent/'.strtolower($this->type).'.stub';
+        }
+
+        return __DIR__.'/stubs/abstract/'.strtolower($this->type).'.stub';
+    }
+
+    /**
+     * Get the default namespace for the class.
+     *
+     * @param  string  $rootNamespace
+     * @return string
+     */
+    protected function getDefaultNamespace($rootNamespace)
+    {
+        return implode('', [
+            $rootNamespace,
+            '\\',
+            $this->subNamespace,
+            '\\',
+            $this->getPluralizedResource()
+        ]);
+    }
+
+    /**
+     * Get the console command arguments.
+     *
+     * @return array
+     */
+    protected function getArguments()
+    {
+        return [
+            ['resource', InputArgument::REQUIRED, 'The resource for which a Schema will be generated'],
+        ];
+    }
+
+    /**
+     * Get the console command options.
+     *
+     * @return array
+     */
+    protected function getOptions()
+    {
+        return [
+            ['eloquent', 'e', InputOption::VALUE_OPTIONAL, 'Use eloquent as adapter.'],
```

Comment:
For the eloquent option, I think have `--eloquent` and `--no-eloquent` as boolean options. If `--eloquent` and `--no-eloquent` both return `false` (i.e. neither have been provided) then fall back to the value in config. I.e...

``` php
protected function isEloquent()
{
	if ($this->isIndependent) {
		return false;
	}
	
	if ($this->option('no-eloquent')) {
		return false;
	}
	
	return $this->option('eloquent') ?: $this->useEloquent;
}
```